crates/ntx-builder/src/state/account.rs

```diff
@@ -153,8 +207,22 @@ impl AccountState {
             && self.available_notes.is_empty()
             && self.nullified_notes.is_empty()
     }
+
+    /// Marks the specified notes as failed.
+    pub fn fail(&mut self, notes: &[InflightNetworkNote], block_num: BlockNumber) {
+        for nullifier in notes.iter().map(|note| note.to_inner().nullifier()) {
+            if let Some(note) = self.available_notes.get_mut(&nullifier) {
+                note.fail(block_num);
+            } else {
+                tracing::warn!("failed to find note with nullifier {nullifier}");
```

Comment:
```suggestion
                tracing::warn!(%nullifier, "failed note is not in account's state");
```

We should also consider how we want to formally trace these event's. This is fine for now; but we should consider what to instrument when we refactor to actors. For example it would be good to know the account ID here which is possible similar to `latest_account`.